### PR TITLE
Refactor uploadProgram and SendMessageToProgram methods

### PR DESCRIPTION
**Pull request by aderrod94**
Bump @gear-js/api on frontend

### CODE REVIEW

File: website/frontend/src/services/ApiService.tsx [hunk 1]
```diff
@@ -152,17 +188,55 @@ export const SendMessageToProgram = async (
...
+
+      if (data.status.isFinalized) {
+        data.events.forEach((event: any) => {
+          const { method } = event.event;
+
+          if (method === 'DispatchMessageEnqueued') {
+            dispatch(
+              AddAlert({
+                type: EventTypes.SUCCESS,
+                message: `Send message status: finalized ${data.status.asFinalized}`,
```
Comment (osipov-mit) [inline website/frontend/src/services/ApiService.tsx:209]:
```suggestion
                message: `Send message: Finalized`,
```

File: website/frontend/src/services/ApiService.tsx [hunk 2]
```diff
@@ -152,17 +188,55 @@ export const SendMessageToProgram = async (
...
+          AddAlert({
+            type: EventTypes.ERROR,
+            message: PROGRAM_ERRORS.INVALID_TRANSACTION,
+          })
+        );
       }
     });
   } catch (error) {
-    dispatch(AddAlert({ type: EventTypes.ERROR, message: `SEND MESSAGE STATUS: ${error}` }));
+    dispatch(AddAlert({ type: EventTypes.ERROR, message: `Send message status: ${error}` }));
```
Comment (osipov-mit) [inline website/frontend/src/services/ApiService.tsx:238]:
```suggestion
    dispatch(AddAlert({ type: EventTypes.ERROR, message: `Send message: ${error}` }));
```

File: website/frontend/src/services/ApiService.tsx [hunk 3]
```diff
@@ -71,69 +71,105 @@ export const UploadProgram = async (
...
-                dispatch(AddAlert({ type: EventTypes.SUCCESS, message: `Metadata added successfully` }));
-              }
-            } catch (error) {
-              dispatch(AddAlert({ type: EventTypes.ERROR, message: `${error}` }));
-              console.error(error);
+      if (data.status.isInBlock) {
+        dispatch(
+          AddAlert({
+            type: EventTypes.SUCCESS,
+            message: `Upload program status: in block ${data.status.asInBlock}`,
```
Comment (osipov-mit) [inline website/frontend/src/services/ApiService.tsx:83]:
```suggestion
            message: `Upload program: InBlock`,
```

File: website/frontend/src/services/ApiService.tsx [hunk 4]
```diff
@@ -71,69 +71,105 @@ export const UploadProgram = async (
...
+
+      if (data.status.isFinalized) {
+        data.events.forEach((event: any) => {
+          const { method } = event.event;
+
+          if (method === 'InitMessageEnqueued') {
+            dispatch(
+              AddAlert({
+                type: EventTypes.SUCCESS,
+                message: `Upload program status: finalized in block ${data.status.asFinalized}`,
```
Comment (osipov-mit) [inline website/frontend/src/services/ApiService.tsx:96]:
```suggestion
                message: `Upload program: Finalized`,
```

File: website/frontend/src/services/ApiService.tsx [hunk 5]
```diff
@@ -71,69 +71,105 @@ export const UploadProgram = async (
...
+                  initStatus: ProgramStatus.Success,
+                  meta: {
+                    meta: JSON.stringify(meta),
+                    metaFile,
+                    programId,
+                  },
+                  uploadedAt: Date(),
+                })
+                .then(() => {
+                  dispatch(AddAlert({ type: EventTypes.SUCCESS, message: `Program added successfully` }));
```
Comment (osipov-mit) [inline website/frontend/src/services/ApiService.tsx:117]: Added where?

File: website/frontend/src/services/ApiService.tsx [hunk 6]
```diff
@@ -71,69 +71,105 @@ export const UploadProgram = async (
...
+                    name,
+                    title,
+                    metaFile,
+                  });
+
+                  if (response.error) {
+                    // FIXME 'throw' of exception caught locally
+                    throw new Error(response.error.message);
+                  } else {
+                    dispatch(AddAlert({ type: EventTypes.SUCCESS, message: `Metadata added successfully` }));
```
Comment (osipov-mit) [inline website/frontend/src/services/ApiService.tsx:139]:
```suggestion
                    dispatch(AddAlert({ type: EventTypes.SUCCESS, message: `Metadata saved successfully` }));
```

File: website/frontend/src/services/ApiService.tsx [hunk 7]
```diff
@@ -71,69 +71,105 @@ export const UploadProgram = async (
...
             }
-          });
-        }
+          }
+
+          if (method === 'ExtrinsicFailed') {
+            dispatch(
+              AddAlert({
+                type: EventTypes.ERROR,
+                message: `Extrinsic Failed`,
```
Comment (osipov-mit) [inline website/frontend/src/services/ApiService.tsx:153]:
```suggestion
                message: `Upload program: Extrinsic Failed`,
```

File: website/frontend/src/services/ApiService.tsx [hunk 8]
```diff
@@ -71,69 +71,105 @@ export const UploadProgram = async (
...
+          })
+        );
       }
     });
   } catch (error) {
     dispatch(programUploadFailedAction(`${error}`));
-    console.error(error);
-    dispatch(AddAlert({ type: EventTypes.ERROR, message: `UPLOAD STATUS: ${error}` }));
-    // alert.error(`status: ${error}`);
+    dispatch(AddAlert({ type: EventTypes.ERROR, message: `Upload status: ${error}` }));
```
Comment (osipov-mit) [inline website/frontend/src/services/ApiService.tsx:172]:
```suggestion
    dispatch(AddAlert({ type: EventTypes.ERROR, message: `Upload program: ${error}` }));
```

**Review by osipov-mit — CHANGES_REQUESTED** [review]

**Review by osipov-mit — APPROVED** [review]

**Review by nikitayutanov — APPROVED** [review]

**Pull request merged**